### PR TITLE
ITEP-93715 - Testset for cluster-analytics: add unit and component test suites

### PR DESCRIPTION
**Pull request by ltalarcz**
Adds two test tiers for cluster analytics service, closing the gap of zero test coverage before the planned tracker refactoring.

Unit tests (tests/sscape_tests/cluster_analytics/):
- test_config.py: config loading, DBSCAN defaults, per-category overrides
- test_clustering.py: DBSCAN grouping, centroids, noise exclusion, fallback
- test_shape.py: shape detection (circle, line, irregular, insufficient)
- test_velocity.py: velocity classification (stationary, parallel, converging, diverging, insufficient)
- test_tracker.py: post-refactor ClusterTracker API spec (skipped due to tracker that is going to be simplified)

Result: 33 passed, 8 skipped (no Docker required)

Component tests (cluster_analytics/tests/service/):
- Spins up isolated Mosquitto + cluster-analytics container per test
- Publishes DATA_REGULATED messages, asserts ANALYTICS_CLUSTERS output
- Covers: single/two groups, UUID persistence, categories, below min_samples, empty input
- Includes warmup helper for FSM FRAMES_TO_ACTIVATE (3 frames) requirement

Result: 6 passed (~15s total)

Also adds cluster_analytics/tests/README.md with run instructions and updates cluster_analytics/README.md with Testing section.

## 📝 Description

Provide a clear summary of the changes and the context behind them. Describe **what** was changed, **why** it was needed, and **how** the changes address the issue or add value.

## ✨ Type of Change

Select the type of change your PR introduces:

- [ ] 🐞 **Bug fix** – Non-breaking change which fixes an issue
- [ ] 🚀 **New feature** – Non-breaking change which adds functionality
- [ ] 🔨 **Refactor** – Non-breaking change which refactors the code base
- [ ] 💥 **Breaking change** – Changes that break existing functionality
- [ ] 📚 **Documentation update**
- [ ] 🔒 **Security update**
- [ ] 🧪 **Tests**
- [ ] 🚂 **CI**

## 🧪 Testing Scenarios

Describe how the changes were tested and how reviewers can test them too:

- [ ] ✅ Tested manually
- [ ] 🤖 Ran automated end-to-end tests

## ✅ Checklist

Before submitting the PR, ensure the following:

- [ ] 🔍 PR title is clear and descriptive
- [ ] 📝 For internal contributors: If applicable, include the JIRA ticket number (e.g., ITEP-123456) in the PR **title**. Do **not** include full URLs
- [ ] 💬 I have commented my code, especially in hard-to-understand areas
- [ ] 📄 I have made corresponding changes to the documentation
- [ ] ✅ I have added tests that prove my fix is effective or my feature works

### CODE REVIEW

File: cluster_analytics/tests/README.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+| `test_config.py` | Config loading, defaults, per-category DBSCAN overrides |
+| `test_clustering.py` | DBSCAN grouping, centroids, noise exclusion, empty inputs |
+| `test_shape.py` | Shape detection: circle, line, irregular, insufficient points |
+| `test_velocity.py` | Velocity classification: stationary, parallel, converging, diverging |
+| `test_tracker.py` | Post-refactor tracker API spec (currently skipped — Phase 1) |
+
+**Run from repo root:**
+
+```bash
+python -m pytest tests/sscape_tests/cluster_analytics/ -v -p no:django
```
Comment (dmytroye) [inline cluster_analytics/tests/README.md:44]: Please add instructions to setup env to run this testset.

Is scikit-learn missed in /tests/requirements.txt?

File: tests/sscape_tests/cluster_analytics/conftest.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+"""
+
+import sys
+import pytest
+from pathlib import Path
+from unittest.mock import MagicMock, patch
+
+_REPO_ROOT = Path(__file__).resolve().parents[3]
+_CLUSTER_SRC = _REPO_ROOT / "cluster_analytics" / "src"
+
```
Comment (dmytroye) [inline tests/sscape_tests/cluster_analytics/conftest.py:20]: Met issue with importing `scene_common` while executing from README.md instructions.
```suggestion
_SCENE_COMMON_SRC = _REPO_ROOT / "scene_common" / "src"

if str(_SCENE_COMMON_SRC) not in sys.path:
  sys.path.insert(0, str(_SCENE_COMMON_SRC))
  # Evict any namespace-package stub so the real package is re-resolved
  sys.modules.pop("scene_common", None)
```

File: cluster_analytics/tests/service/test_clustering_pipeline.py [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    assert "vehicle" in categories
+    # Confirm no category mixing: each cluster's object_ids all start with category prefix
+    for cluster in msg["clusters"]:
+      prefix = cluster["category"]
+      for oid in cluster["object_ids"]:
+        assert oid.startswith(prefix), (
+          f"Cluster of '{prefix}' contains id '{oid}' from another category"
+        )
+
+  def test_below_min_samples_produces_empty_clusters(self, mqtt_client, cluster_analytics_service):
```
Comment (dmytroye) [inline cluster_analytics/tests/service/test_clustering_pipeline.py:148]: Let's add a negative test case for empty clusters produced when objects to far away from each other.

**Review by dmytroye — APPROVED** [review]
